nameko/messaging.py

```diff
@@ -310,12 +308,18 @@ def unregister_provider(self, provider):
 
         super(QueueConsumer, self).unregister_provider(provider)
 
-    @retry(for_exceptions=RecoverableConnectionError, max_attempts=3)
     def ack_message(self, message):
+        if not message.channel.connection:
```

Comment:
don't you have a race here? could we try and catch+ignore the exception?

Reply:
Race between what?

Try/catch is problematic because the exception raised is `RecoverableConnectionError`, which "connection already closed" just as a string. It's worth noting that the connection is not actually recoverable from here (we used to retry on these errors, and even after a new connection is established the ack is still attempted through the dead channel) and even if it _was_, the broker will already have reclaimed the message.

The logic here is "if the channel is closed, the message will have been reclaimed, so do nothing".

Reply:
> Race may be wrong word, but connection might close between checking and acking

(GitHub email didn't reply to my comment)

Yes I am with you. I would like to keep the conditional to show intent, but will add a try/catch to make sure we don't blow up if we lose the race.